python/lsst/ts/salobj/topics/read_topic.py

```diff
@@ -215,35 +215,36 @@ class ReadTopic(BaseTopic):
     -----
     **Queues**
 
-    There are actually two queues: a Python queue whose length
-    is set by ``queue_len`` and a dds queue whose length is set by
-    the DDS QoS file. Data can be lost in two ways:
+    There are two queues: a Python queue whose length is set by ``queue_len``
+    and a dds queue whose length is set by the DDS Quality of Service file.
+    In the doc strings for the methods, below, any reference to the queue
+    refers to the Python queue.
+
+    Data can be lost from either queue:
 
     - If this class cannot read messages from the DDS queue fast enough,
       then older messages will be dropped from the DDS queue. You will get
-      a warning log message if the reader starts to fall behind.
-    - As messages are read they are put on the Python queue.
+      several warning log messages as the DDS queue fills.
+    - As messages are read from the DDS queue they are put on the Python queue.
       If a callback function or `next` does not process data quickly enough
       then older messages are dropped from the Python queue.
       If you have a callback function then you will get several
-      warning log messages as this Python queue fills up.
-      You get no warning otherwise because this class has no way of knowing
+      warning log messages as the Python queue fills up;
+      you get no warning otherwise because `ReadTopic` has no way of knowing
       whether or not you intend to read all messages.
 
     **Reading**
 
-    Reading is performed by the `.SalInfo` which has single read loop that
-    reads messages for all topics. This is more efficient than having each
-    `ReadTopic` read its own messages.
+    Reading is performed by the contained `SalInfo`, which has single
+    read loop that reads messages for all topics. This is more efficient
+    than having each `ReadTopic` read its own messages.
 
     **Modifying Messages**
 
-    All functions that return messages return from an internal cache.
-    This cached value is replaced by all read operations. So as long as
-    no other code modifies a message, the returned message will not change.
-
-    To safely modify a returned message, make your own copy
-    with ``copy.copy(data)``.
+    All functions that return messages return them from an internal cache.
```

Comment:
How is the cache different from the Python queue itself?

Reply:
There are at least two sources of cached data: the Python queue and also the `_current_data` attribute.